src/PyNvCodec/src/PyNvEncoder.cpp

```diff
@@ -177,7 +198,8 @@ bool PyNvEncoder::EncodeSingleSurface(EncodeContext& ctx)
         cuda_str, cuda_ctx, cli_interface,
         NV12 == eFormat ? NV_ENC_BUFFER_FORMAT_NV12
                         : YUV444 == eFormat ? NV_ENC_BUFFER_FORMAT_YUV444
-                                            : NV_ENC_BUFFER_FORMAT_UNDEFINED,
+                        : YUV444_10bit == eFormat ? NV_ENC_BUFFER_FORMAT_YUV444_10BIT
+                        : NV_ENC_BUFFER_FORMAT_UNDEFINED,
```

Comment:
I would prefer to move this decision to a non inline variable using a switch  statement. This is not very readable and error prone.